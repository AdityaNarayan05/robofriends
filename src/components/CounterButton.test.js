import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

describe('CounterButton', () => {
  it('should render CounterButton component', () => {
    const wrapper = shallow(<CounterButton color="blue" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should increment count when button is clicked', () => {
    const wrapper = shallow(<CounterButton color="blue" />);
    const button = wrapper.find('button');

    expect(wrapper.state('count')).toEqual(1);

    button.simulate('click');
    expect(wrapper.state('count')).toEqual(2);

    button.simulate('click');
    expect(wrapper.state('count')).toEqual(3);
  });

  it('should not update component when count state remains the same', () => {
    const wrapper = shallow(<CounterButton color="blue" />);
    const shouldUpdate = wrapper.instance().shouldComponentUpdate({ color: 'red' }, { count: 1 });

    expect(shouldUpdate).toBe(false);
  });

  it('should update component when count state changes', () => {
    const wrapper = shallow(<CounterButton color="blue" />);
    const shouldUpdate = wrapper.instance().shouldComponentUpdate({ color: 'red' }, { count: 2 });

    expect(shouldUpdate).toBe(true);
  });
});
