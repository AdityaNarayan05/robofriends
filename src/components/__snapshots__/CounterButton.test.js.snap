// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton should render CounterButton component 1`] = `
<button
  color="blue"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
